core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -384,11 +431,38 @@ private void configureConnectors(SslContextFactory sslContextFactory) {
     for (URI listener : listeners) {
       log.info("Adding listener: " + listener.toString());
       NetworkTrafficServerConnector connector;
-      if (listener.getScheme().equals("http")) {
-        connector = new NetworkTrafficServerConnector(this, httpConnectionFactory);
+
+      // Default to supporting HTTP/2 for Java 11 and later
+      if (IS_JAVA11_COMPATIBLE) {
+        if (listener.getScheme().equals("http")) {
+          // HTTP2C is HTTP/2 Clear text
+          final HTTP2CServerConnectionFactory h2cConnectionFactory =
+                  new HTTP2CServerConnectionFactory(httpConfiguration);
+  
+          // The order of HTTP and HTTP/2 is significant here but it's not clear why :)
+          connector = new NetworkTrafficServerConnector(this, null, null, null, 0, 0,
+                  httpConnectionFactory, h2cConnectionFactory);
+        } else {
+          final HTTP2ServerConnectionFactory h2ConnectionFactory =
+                  new HTTP2ServerConnectionFactory(httpConfiguration);
+  
+          ALPNServerConnectionFactory alpnConnectionFactory = new ALPNServerConnectionFactory();
+          alpnConnectionFactory.setDefaultProtocol(HttpVersion.HTTP_1_1.asString());
+  
+          SslConnectionFactory sslConnectionFactory = new SslConnectionFactory(sslContextFactory,
+                  alpnConnectionFactory.getProtocol());
+  
+          connector = new NetworkTrafficServerConnector(this, null, null, null, 0, 0,
+                  sslConnectionFactory, alpnConnectionFactory, h2ConnectionFactory,
+                  httpConnectionFactory);
+        }
       } else {
-        connector = new NetworkTrafficServerConnector(this, httpConnectionFactory,
-                sslContextFactory);
+        if (listener.getScheme().equals("http")) {
+          connector = new NetworkTrafficServerConnector(this, httpConnectionFactory);
+        } else {
+          connector = new NetworkTrafficServerConnector(this, httpConnectionFactory,
+                  sslContextFactory);
+        }
```

Comment:
nit: What do you think about extracting this in its own method?

Reply:
@dimitarndimitrov I think that the `configureConnectors` method doesn't do a lot more than this anyway. Also, the configuration of the connectors isn't used anywhere else. Personally, I don't think there's a heap of value in extracting this into its own method at this point.

Reply:
+1 to @dimitarndimitrov suggestion. Having such a long body inside a for-loop decreases readability a little bit. I would extract the whole body of the for-loop out. I would also in addition possibly extract the body of each branch of the conditional in its own function.

Reply:
for-loop body refactored into own function.